beta-operator-rewards/rewards-requirements.ts

```diff
@@ -0,0 +1,229 @@
+import { Contract } from "ethers";
+import { program } from "commander";
+import * as fs from "fs";
+import { ethers } from "ethers";
+import {
+  abi as RandomBeaconABI,
+  address as RandomBeaconAddress,
+} from "@keep-network/random-beacon/artifacts/RandomBeacon.json";
+import {
+  abi as WalletRegistryABI,
+  address as WalletRegistryAddress,
+} from "@keep-network/ecdsa/artifacts/WalletRegistry.json";
+import axios from "axios";
+
+export async function calculateBetaOperatorFactors() {
+  program
+    .version("0.0.1")
+    .requiredOption(
+      "-s, --start <timestamp>",
+      "starting time for rewards calculation"
+    )
+    .requiredOption(
+      "-e, --end <timestamp>",
+      "ending time for rewards calculation"
+    )
+    .requiredOption("-i, --interval <timestamp>", "scrape interval") // IMPORTANT, it has to be the same as in prometheous config
+    .requiredOption(
+      "-a, --api <prometheus api>",
+      "prometheus API"
+    )
+    .requiredOption(
+      "-j, --job <prometheus job>",
+      "prometheus job"
+    )
+    .requiredOption(
+      "-o, --output <file>",
+      "output JSON file"
+    )
+    .parse(process.argv);
+
+  const options = program.opts();
+
+  const queryStep = 600; // 10min in sec
+  const prometheous_job = options.job;
+  const prometheusAPI = options.api
+  const startRewardsTimestamp = parseInt(options.start);
+  const endRewardsTimestamp = parseInt(options.end);
+  const scrapeInterval = parseInt(options.interval);
+  const rewardsInterval = endRewardsTimestamp - startRewardsTimestamp;
+  const requiredUptime = 96; // percent
+  const preParamsAvgInterval = 24 // hours
+  const preParamsResolution = 24 // hours
+  const minPreParams = 500 // min requirement for pre params daily avg
+  const factors = {
+    isBeaconAuthorized: "isBeaconAuthorized",
+    isTbtcAuthorized: "isTbtcAuthorized",
+    upTime: "upTime",
+    version: "verion",
```

Comment:
```suggestion
    version: "version",
```

Reply:
Fixed as part of [982a1f0](https://github.com/keep-network/keep-core/pull/3346/commits/982a1f01cbd3036fa4c035d697edb323b75493c6)